### PR TITLE
Docs: Improve upgrade guide

### PR DESCRIPTION
**Pull request by Rachael-Graham**
Improve the upgrade guide into a more cohesive experience, including rolling the version-specific info into the upgrade steps themselves so users dont miss those.
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/8241

### CODE REVIEW

File: docs/content/operations/upgrading/upgrade_steps.md [hunk 1]
```diff
@@ -65,6 +37,70 @@ You might deploy Gloo Edge in Kubernetes environments that use the Kubernetes lo
 Try a [Canary upgrade]({{< versioned_link_path fromRoot="/operations/upgrading/canary" >}}) to make sure that the newer version works as you expect before upgrading.
 {{% /notice %}}
 
+## Step 2: Review version {{< readfile file="static/content/version_geoss_latest_minor.md" markdown="true">}} changes
```
Comment (Rachael-Graham) [inline docs/content/operations/upgrading/upgrade_steps.md:40]: check if this renders

File: docs/content/operations/upgrading/upgrade_steps.md [hunk 2]
```diff
@@ -65,6 +37,70 @@ You might deploy Gloo Edge in Kubernetes environments that use the Kubernetes lo
...
+     ~ > glooctl version
+     Client: {"version":"{{< readfile file="static/content/version_geoss_latest.md" markdown="true">}}"}
+     Server: {"type":"Gateway","enterprise":true,"kubernetes":...,{"Tag":"{{< readfile file="static/content/version_gee_latest.md" markdown="true">}}","Name":"grpcserver-ee","Registry":"quay.io/solo-io"},...,{"Tag":"{{< readfile file="static/content/version_geoss_latest.md" markdown="true">}}","Name":"discovery","Registry":"quay.io/solo-io"},...}
+     ```
+
+2. Review the following main feature, Helm, CRD, and CLI changes for version {{< readfile file="static/content/version_geoss_latest_minor.md" markdown="true">}}.
+
+3. If you still aren't sure about the version upgrade impact, scan our [Frequently-asked questions]({{% versioned_link_path fromRoot="/operations/upgrading/faq/" %}}). Also, feel free to post in the `#gloo` or `#gloo-enterprise` channels of our [public Slack](https://slack.solo.io/) if your use case doesn't quite fit the standard upgrade path.
+
+### Feature changes {#features}
```
Comment (Rachael-Graham) [inline docs/content/operations/upgrading/upgrade_steps.md:57]: these are all just placeholders for now until it is time to actually fill them out in prep for 1.15 release

File: docs/content/operations/upgrading/upgrade_steps.md [hunk 3]
```diff
@@ -131,17 +160,62 @@ You can use the `glooctl upgrade` command to download the latest binary. For mor
...
+      ```sh
+      helm repo update
+      helm pull glooe/gloo-ee --version 1.15.0 --untar
+      kubectl apply -f gloo-ee/charts/gloo/crds
+      # If Gloo Federation is enabled
+      kubectl apply -f gloo-ee/charts/gloo-fed/crds
+      ```
+      {{% /tab %}}
+      {{< /tabs >}}
+   2. Verify that the deployed CRDs use the same version as your current Gloo Edge installation.
```
Comment (Rachael-Graham) [inline docs/content/operations/upgrading/upgrade_steps.md:195]: This was the existing text in the guide, but this doesnt seem correct... shouldnt the crds run the upgrade version?

File: docs/content/operations/upgrading/canary.md [hunk 4]
```diff
@@ -1,199 +1,240 @@
...
-   * Places the federated virtual service resource in both the `remote1` old version cluster (1.12) and the `remote2` new target version cluster (1.13).
+8. Check the [Feature changes]({{% versioned_link_path fromRoot="/operations/upgrading/faq/#features" %}}) for any changes or new capabilities in {{< readfile file="static/content/version_geoss_latest_minor.md" markdown="true">}}. In your management cluster, [create or modify any federated custom resources]({{< versioned_link_path fromRoot="/guides/gloo_federation/federated_configuration/" >}}) with these changes to test your canary version of Gloo Edge Federation. Because Gloo Edge Federation scans all federated resources on the management cluster, you might want to reuse the existing federated resources in the `gloo-system` namespace. <!--If applicable, add steps to walk users though updating crs for any breaking changes--> 
+   {{% expand "Expand for an example federated virtual service configuration" %}}
+   This example configuration creates a federated virtual service with different features across versions 1.12 and 1.13.<!--IF POSSIBLE UPDATE for something that is new in 1.15 instead-->
```
Comment (Rachael-Graham) [inline docs/content/operations/upgrading/canary.md:154]: So the example they give of different config across 2 minor version is a VS in 1.13, bc that resource had new options. I would like to update this with a 1.15 and 1.14 set instead but i guess will need to wait and see what new config optins are avail in 1.15 to demonstrate

**Review by Nadine2016 — APPROVED** [review]

**Pull request merged**